pywisp/widgets/fileselector.py

```diff
@@ -14,23 +16,38 @@ def __init__(self, formats):
         super().__init__()
         self.st = settings(_default)
         self.formats = formats
+        self.exts = [self._extFromFilter(flt) for flt in formats]
+        self.logger = logging.getLogger("FileSelector")
+
+    def _extFromFilter(self, flt):
+        return re.search(r'\*([.a-z]*)', flt).group(1)
 
     def getSaveFileName(self, text="Export as ..."):
         path = self.st.value("path/export_dir")
         ext = self.st.value("path/export_ext")
-        file = os.path.join(path, "export" + ext)
+        file = os.path.join(path, "export")
         filterStr = ";;".join(self.formats)
         ix = [ ext in fmt for fmt in self.formats ].index(True)
 
-        selection = super().getSaveFileName(self, text, file, filterStr, self.formats[ix])
-        if not (filename := selection[0]):
+        selpath, selfilt = super().getSaveFileName(self, text, file,
+                                                   filterStr, self.formats[ix])
+        if not selpath:
+            self.logger.warn("File selection aborted!")
+            return
+        name, selext = os.path.splitext(selpath)
+        if not selext:
+            selext = self._extFromFilter(selfilt)
+            selpath += selext
+        elif selext not in selfilt:
+            self.logger.error(f"Extension '{selext}' doesn't match used filter. Aborting!")
```

Comment:
I know that this will rarely happen anyway but if an extension is given, I think we really should not care which filter was selected in the file dialog while doing so. As long as the extension is supported by the exporter, of course.